src/include/ASTree.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <Token.h>
+#include <vector>
+#include <ios>
+#include <sstream>
+
+struct Statement { };
+
+//////////////////////////////////////////
+/// Expressions
+
+struct Expression : public Statement { };
+
+class ExpressionGroup : public Expression {
+    std::unique_ptr<Expression> m_expression;
+public:
+    explicit ExpressionGroup(std::unique_ptr<Expression> expression)
+            : m_expression(std::move(expression)) { }
+};
+
+class BinaryExpression : public Expression {
+    TokenType m_op;
+    std::unique_ptr<ExpressionGroup> m_left;
+    std::unique_ptr<ExpressionGroup> m_right;
+public:
+    BinaryExpression(TokenType op, std::unique_ptr<ExpressionGroup> left,
+                     std::unique_ptr<ExpressionGroup> right)
+            : m_op(op)
+            , m_left(std::move(left))
+            , m_right(std::move(right)) { }
+};
+
+class UnaryExpression : public Expression {
+    char m_op;
+    std::unique_ptr<ExpressionGroup> m_operand;
+public:
+    UnaryExpression(char op, std::unique_ptr<ExpressionGroup> operand)
+            : m_op(op)
+            , m_operand(std::move(operand)) { }
+};
+
+class FunctionCall : public Expression {
+    std::string m_functionName;
+    std::vector<std::unique_ptr<ExpressionGroup>> m_args;
```

Comment:
Why a vector of expression groups in a function call? Shouldn't it be a vector of expressions and/or identifiers?

Reply:
hm you're right. I'm not actually sure how to handle IDENT yet - will look into is soon, but this will probably be `Expression`